pkg/render/node.go

```diff
@@ -283,6 +304,38 @@ func (c *nodeComponent) nodeRoleBinding() *rbacv1.ClusterRoleBinding {
 	return crb
 }
 
+// nodeRoleBinding creates a clusterrolebinding giving the node service account the required permissions to operate.
+func (c *nodeComponent) CNIPluginRoleBinding() *rbacv1.ClusterRoleBinding {
+	finalizer := []string{}
+	if !c.cfg.Terminating {
+		finalizer = []string{NodeFinalizer}
+	}
+	crb := &rbacv1.ClusterRoleBinding{
+		TypeMeta: metav1.TypeMeta{Kind: "ClusterRoleBinding", APIVersion: "rbac.authorization.k8s.io/v1"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:       CalicoCNIPluginObjectName,
+			Labels:     map[string]string{},
+			Finalizers: finalizer,
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "ClusterRole",
+			Name:     CalicoCNIPluginObjectName,
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:      "ServiceAccount",
+				Name:      CalicoCNIPluginObjectName,
+				Namespace: common.CalicoNamespace,
+			},
+		},
+	}
+	if c.cfg.MigrateNamespaces {
+		migration.AddBindingForKubeSystemNode(crb)
```

Comment:
I don't think this is doing the right thing - this is binding this role to the `calico-node` serviceaccount in the kube-system namespace.

What we want to do is bind it to the `calico-cni-plugin` service account in the kube-system namespace. LMK if it's not clear why - the migration logic isn't necessarily obvious.